EEJT-0002/src/main/java/com/epam/esm/exceptionhandler/handler/ResponseExceptionHandler.java

```diff
@@ -0,0 +1,38 @@
+package com.epam.esm.exceptionhandler.handler;
+
+import com.epam.esm.exceptionhandler.exception.ItemNotFoundException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.sql.SQLSyntaxErrorException;
+import java.util.Map;
+
+@ControllerAdvice
+public class ResponseExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(value = {IllegalArgumentException.class, IllegalStateException.class})
+    protected ResponseEntity<?> handleConflictException(RuntimeException ex, WebRequest request) {
+        return handleExceptionInternal(ex,
+                Map.of("HTTP Status", HttpStatus.CONFLICT, "response body", Map.of("error message", ex.getMessage())),
```

Comment:
you throw IllegalArgumentException when client send request with invalid data, its' not the 409 CONFLIT status, it's 400. Server may return 409, when client try to update resource which have the newest version than in client request or for example specific business logic for tickets to cinema, train

Reply:
fix it